Fix multiselect fields

edlerd opened the pull request:
## Done

- Fix multiselect fields, thanks to @kimanhou for enabling this with https://github.com/canonical/react-components/pull/1258
- Connect label to input
- Add help text to input

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @Kxiru or @edlerd for access.
    - With a local copy of this branch, [build and run as described in the docs](https://github.com/canonical/lxd-ui/blob/main/CONTRIBUTING.md#setting-up-for-development).
2. Perform the following QA steps:
    - edit or create a project and enable custom restrictions, then ensure under restrictions > cluster when overriding "Cluster group targeting", that the label is connected to the input and that the help text is visible below the input.
    - Under profile/instance > configuration > other/proxy device. Create a device of each type and ensure the help text renders fine
    - Under profile/instance > configuration > resource limits/security policies/snapshots/migration enable overrides and ensure the help text renders fine
    - Under Settings create overrides for several server settings and ensure the help text renders fine

## Screenshots

<img width="1594" height="957" alt="image" src="https://github.com/user-attachments/assets/df9d852a-cc9c-4bc3-b776-77d6d5185384" />

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://lxd-ui-1520.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/24641)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

kimanhou commented:
QA done, looks nice !
<img width="1017" height="424" alt="Screenshot from 2025-10-03 18-17-24" src="https://github.com/user-attachments/assets/ac619b76-d06a-4eba-b643-a9e3683a2829" />

kimanhou commented:
There is an error in the console: `In HTML, <p> cannot be a descendant of <p>.`
Indeed, there are 2 nested paragraphs.
<img width="874" height="112" alt="image" src="https://github.com/user-attachments/assets/d00cc385-cd1f-4f70-b86e-9d9c6e6d4f93" />
This is because `ConfigFieldDescription` renders a paragraph.

edlerd commented:
Fixed the `p` not appearing as a descendant of another paragraph as a drive by. Updated QA steps accordingly, as this PR now affects more places in the UI.
